jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -160,14 +167,16 @@ public static exportResult exportJar(List<Object> arguments, IProgressMonitor mo
         }
         try (JarOutputStream target = new JarOutputStream(new FileOutputStream(destination), manifest)) {
             Set<String> directories = new HashSet<>();
-            for (String classpath : classpaths) {
-                if (classpath != null) {
-                    if (classpath.endsWith(".jar")) {
-                        ZipFile zip = new ZipFile(classpath);
-                        writeArchive(zip, true, true, target, directories, monitor);
+            for (ClassPath classpath : classpaths) {
+                if (classpath.isExtract) {
+                    ZipFile zip = new ZipFile(classpath.source);
+                    writeArchive(zip, true, true, target, directories, monitor);
+                } else {
+                    File file = new File(classpath.source);
+                    if (StringUtils.isEmpty(classpath.destination)) {
```

Comment:
why we need `writeFileRecursively` when `destination` is empty?

Reply:
When we use the wizard to export jar. We will export a series of folders containing class file. `writeFileRecursively` handle these folders.
When we use a custom task to export jar, we will get the file list (not include any directory) from globby.  We use a trie to save and specific each destination of a particular file, so the `ClassPath` has a valid `destination`. So we simply call `writeFile`.
Do you think it's better to merge the behavior of them and use globby to get the file list when we use the wizard to export jar?